jupyterlab_git/handlers.py

```diff
@@ -73,7 +73,10 @@ async def post(self, path: str = ""):
         """
         data = self.get_json_body()
         response = await self.git.clone(
-            self.url2localpath(path), data["clone_url"], data.get("auth", None)
+            self.url2localpath(path),
+            data["clone_url"],
+            data.get("auth", None),
+            data.get("cache_credentials", False),
```

Comment:
Let's propagate the extended auth object from the frontend directly (i.e. `cache_credentials` will be a new key in `auth`)

The reason being it is that the credentials caching without authentication information has no effect.
Could you also propagate that to the Git clone, push, ... methods.